### PR TITLE
Features/label color combine

### PR DESCRIPTION
**Pull request by tmarkn**
Added color changing

### CODE REVIEW

File: static/js/app.js [hunk 1]
```diff
@@ -3,106 +3,158 @@
...
+      html: '.hwt-content .label_rem {background-color: rgb(255, 192, 203);}'
+    })
+  );
+  $('head').append(
+    $('<style/>', {
+      id: 'sit-style',
+      html: '.hwt-content .label_sit {background-color: rgb(173, 255, 47);}'
+    })
+  );
+}
```
Comment (ghost) [inline static/js/app.js]: You should put the selector in a variable to improve performance. See: https://codereview.stackexchange.com/questions/70298/jquery-duplicated-selector

File: static/js/app.js [hunk 2]
```diff
@@ -3,106 +3,158 @@
...
+    "startPosition": start,
+    "endPosition": end
+  }
 };
 
-function selectedInputRangeIsValid(range){
-  console.log(range.startPosition, range.endPosition);
-    return range.startPosition !== range.endPosition;
+function selectedInputRangeIsValid(range) {
+  console.log(range.startPosition, range.endPosition);              //remove when done
```
Comment (ghost) [inline static/js/app.js]: We should keep the logs but enrich them with more descriptive data i.e. the function it was called in and what variables are being returned ("startposition = " + range.startPosition + "\n" + "endposition = " + range.endPosition)

File: static/js/app.js [hunk 3]
```diff
@@ -3,106 +3,158 @@
...
+    highlight: highlights
+  });
 };
 
 //change the document's label context
-$('input[type=radio]').change(function () {
-    currentLabel = this.value;
+$('.label').on('click', function () {
+  console.log(this.getAttribute('value'));                        //remove when done
+  console.log($(this).css('background-color'));                   //remove when done
```
Comment (ghost) [inline static/js/app.js]: Again we want to keep these but enrich

File: static/js/app.js [hunk 4]
```diff
@@ -3,106 +3,158 @@
...
+  $(this).parent().children('input').click();
+});
+
+//change label color on 
+$('.colorChangePicker').on('change', function () {
+  console.log(this.value);                                        //remove when done
+  console.log($(this).parent()[0].getAttribute('value'));         //remove when done
+  var parent = $(this).parent()[0];
+  $(this).parent().css('background-color', this.value);
+  $('#' + $(this).parent()[0].getAttribute('value') + '-style').html('.hwt-content .label_' + $(this).parent()[0].getAttribute('value') + ' {background-color: ' + this.value + ';}');
```
Comment (ghost) [inline static/js/app.js]: Do we still need the highlightWithinTextArea() plugin with all this is going on?

File: static/js/classes/doc.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ class Doc {
     this.annotations = [];
   }
 
-   getAnnotationRanges(){
-    return this.annotations.map(function(annotation){return annotation.getRangeAsArray();});
+  getAnnotationRanges() {
+    return this.annotations.map(function (annotation) { return annotation.getRangeAsArray(); });
```
Comment (ghost) [inline static/js/classes/doc.js:11]: This can be removed I believe

**Review by ghost — CHANGES_REQUESTED** [review]

File: static/js/app.js [hunk 6]
```diff
@@ -3,106 +3,158 @@
...
+});
+
+//update height on window resize
+$(window).on('resize', function () {
+  $('#doc-view').css('height', 'auto');
+  $('#doc-view').height($('#doc-view').prop('scrollHeight'));
 });
+//calling this twice fixes an issue with height
+$('#doc-view').height($('#doc-view').prop('scrollHeight'));
+$('#doc-view').height($('#doc-view').prop('scrollHeight'));
```
Comment (ghost) [inline static/js/app.js]: Can also be put in a variable to improve performance

File: static/js/app.js (hunk 1 above)
Comment (tmarkn) [reply static/js/app.js]: updated for better performance + less redundancy

File: static/js/app.js (hunk 2 above)
Comment (tmarkn) [reply static/js/app.js]: updated console logging to be more descriptive

File: static/js/app.js (hunk 3 above)
Comment (tmarkn) [reply static/js/app.js]: updated console logging to be more descriptive x2

File: static/js/app.js (hunk 4 above)
Comment (tmarkn) [reply static/js/app.js]: This only changes the colors that they use. It doesn't actually highlight.
\+ Updated for more readability and less redundancy

File: static/js/classes/doc.js (hunk 5 above)
Comment (tmarkn) [reply static/js/classes/doc.js:11]: I'm just going to leave it here in case we need it later

File: static/js/app.js (hunk 6 above)
Comment (tmarkn) [reply static/js/app.js]: updated for better performance + less redundancy x2

**Review by ghost — APPROVED** [review]

**Review by keithudock — APPROVED** [review]

**Review by tuj52136 — APPROVED** [review]

**Pull request merged**